tests/test_stream_manager.c

```diff
@@ -990,12 +1040,159 @@ TEST_CASE(h2_sm_acquire_stream_stress) {
+    size_t num_to_acquire = 1000;
+    size_t accpected_error = 100;
+    size_t num_headers_to_make = 50;
+    size_t header_count = 0;
+    size_t error_count = 0;
+
+    for (size_t i = 0; i < num_to_acquire; i++) {
+        struct aws_http_message *request = aws_http2_message_new_request(s_tester.allocator);
+        ASSERT_NOT_NULL(request);
+        aws_http_message_add_header_array(request, request_headers_src, AWS_ARRAY_SIZE(request_headers_src));
+        /* TODO: why we don't have a message_add_headers? I'll add one... */
+        struct aws_http_headers *request_headers = aws_http_message_get_headers(request);
+        struct aws_http_headers *test_headers = aws_http_headers_new(allocator);
+        for (size_t j = 0; j < num_headers_to_make; j++) {
+            char test_header_str[256];
+            /**
+             * - Don't use _ or - between word, the response will capitalize the first char of every word :(
+             * - Don't use _, the response will change it to -. :(
+             * TODO: maybe not check the header name as only the name will be changed? Maybe have a helper to transform
+             * the header name?
+             */
+            sprintf(test_header_str, "awscommonruntimetest-%zu", header_count++);
+            aws_http_headers_add(
+                test_headers, aws_byte_cursor_from_c_str(test_header_str), aws_byte_cursor_from_c_str(test_header_str));
+            aws_http_headers_add(
+                request_headers,
+                aws_byte_cursor_from_c_str(test_header_str),
+                aws_byte_cursor_from_c_str(test_header_str));
+        }
+        struct aws_byte_buf echo_body;
+        ASSERT_SUCCESS(aws_byte_buf_init(&echo_body, allocator, 100));
+        struct aws_http_make_request_options request_options = {
+            .self_size = sizeof(request_options),
+            .request = request,
+            .user_data = &echo_body,
+            .on_response_body = s_sm_tester_on_echo_body,
+            .on_complete = s_sm_tester_on_stream_complete,
+        };
+        struct aws_http2_stream_manager_acquire_stream_options acquire_stream_option = {
+            .options = &request_options,
+            .callback = s_sm_tester_on_stream_acquired,
+            .user_data = &s_tester,
+        };
+        aws_http2_stream_manager_acquire_stream(s_tester.stream_manager, &acquire_stream_option);
+        /* Wait for the stream to complete */
+        ASSERT_SUCCESS(s_wait_on_streams_completed_count(1));
+        --s_tester.stream_completed_count;
+        if (s_tester.stream_completed_error_code || s_tester.error_code ||
+            !s_echo_body_has_headers(&echo_body, test_headers)) {
+            /* If error happens, we make sure it's acptiable */
+            ++error_count;
+        }
+
+        aws_http_message_release(request);
+        aws_http_headers_release(test_headers);
+        aws_byte_buf_clean_up(&echo_body);
+    }
 
+    ASSERT_TRUE(error_count < accpected_error);
```

Comment:
we're trying to detect subtle compression errors, but we have an "acceptable" error level?

I don't approve of this. Like, I could see allowing some number of requests to fail ... because internet
but allowing a number of string-matches to fail seems like ... why write this test at all?